File: csharp/BronKerbosch/Vertex.cs
using System;
using Index = System.Int32;

#pragma warning disable CA1036 // should define operator(s) '<, <=, >, >=' since it implements IComparable

namespace BronKerbosch
{
    public readonly record struct Vertex : IComparable<Vertex>
    {
        private Index Idx { init; get; }

        public static Vertex Nth(Index idx) => new() { Idx = idx };
        public int CompareTo(Vertex other) => Idx.CompareTo(other.Idx);
        public Index Index() => Idx;
        public override string ToString() => $"vertex {Idx}";
    }
}

File: fsharp/BronKerbosch/Vertex.fs
namespace BronKerbosch

[<StructuredFormatDisplay("v{index}")>]
type public Vertex =
    | Vertex of int

    member this.index = let (Vertex v) = this in v


module Verticise =
    let it (index: int) : Vertex = Vertex.Vertex index
    let first<'T> ((index, arg2): int * 'T) : Vertex * 'T = Vertex.Vertex index, arg2
